### ISSUE TITLE
eslint-disable-line max-len is not working, still throwing a linting error when in use

### ISSUE THREAD

**Issue by kieckhafer** [eslint/eslint#9888]
This is the same issues at #2074 , however that issue has been closed and on that issue you were asking for a new bug report, so here it is.

* **ESLint Version:** 4.16.0
* **Node Version:** 9.4.0
* **npm Version:** 5.6.0

Location of error: https://github.com/reactioncommerce/reaction/blob/fix-3566-kieckhafer-eslintRuleUpdates1/imports/plugins/included/product-variant/components/productGridItems.js#L69

**What did you do? Please include the actual source code causing the issue.**
I'm trying to disable a `max-len` eslint issue on a particular line. The disabling is working at other places in the application, but does not work here:
```
renderAdditionalMedia() {
    if (this.props.additionalMedia() !== false) {
      if (this.props.isMediumWeight()) {
        return (
          <div className={`product-additional-images ${this.renderVisible()}`}>
            {this.props.additionalMedia().map((media) => <span key={media._id} className="product-image" style={{ backgroundImage: `url('${media.url({ store: "medium" })}')` }} />)} // eslint-disable-line max-len
            {this.renderOverlay()}
          </div>
        );
      }
    }
  }
```

**What did you expect to happen?**
I expect that there will not be an error thrown for the line that the `// eslint-disable-lin max-len` is on

**What actually happened? Please include the actual, raw output from ESLint.**
I receive an eslint warning: `69:1  warning  Line 69 exceeds the maximum line length of 160  max-len`

**Comment by not-an-aardvark** [comment]
The issue is that JSX does not allow line comments, so your `// eslint-disable-line` is actually not a comment and it's being rendered as a regular part of your JSX text.

To fix the issue, you can do something like this:

```js
renderAdditionalMedia() {
    if (this.props.additionalMedia() !== false) {
      if (this.props.isMediumWeight()) {
        return (
          <div className={`product-additional-images ${this.renderVisible()}`}>
            {/* eslint-disable max-len */}
            {this.props.additionalMedia().map((media) => <span key={media._id} className="product-image" style={{ backgroundImage: `url('${media.url({ store: "medium" })}')` }} />)}
            {/* eslint-enable max-len */}
            {this.renderOverlay()}
          </div>
        );
      }
    }
  }
```

**Comment by kieckhafer** [comment]
👍 yep, totally correct. Thanks.
